src/GovUk.Education.ExploreEducationStatistics.Admin/Services/FileStorageService.cs

```diff
@@ -177,6 +201,36 @@ public class FileStorageService : IFileStorageService
                 });
         }
 
+        public Task<Either<ActionResult, Release>> CopyReleaseFilesAsync(Guid originalReleaseId, Guid newReleaseId, ReleaseFileTypes type)
+        {
+            return _persistenceHelper
+                .CheckEntityExists<Release>(originalReleaseId)
+                .OnSuccess(_userService.CheckCanViewRelease)
+                .OnSuccess(async newRelease =>
+                {
+                    var blobContainer = await GetCloudBlobContainer();
+
+                    var originalBlobs = blobContainer
+                        .ListBlobs(AdminReleaseDirectoryPath(originalReleaseId, type), true, BlobListingDetails.Metadata)
+                        .Where(blob => !IsBatchedDataFile(blob, originalReleaseId))
+                        .OfType<CloudBlockBlob>()
+                        .ToList();
+
+                    var copyJobs = originalBlobs
+                        .Select(blob =>
+                        {
+                            var originalFilename = blob.Name.Substring(blob.Name.LastIndexOf('/') + 1);
+                            var blobCopy = blobContainer
+                                .GetBlockBlobReference(AdminReleasePath(newReleaseId, type, originalFilename));
+                            return blobCopy.StartCopyAsync(blob);
```

Comment:
In FileStorage service of the Publisher we use TransferManager to do this copy from one path to another. There is a bit more code to set up all of the callbacks for setting what should be excluded and logging when a transfer fails or is skipped, but it might be more reliable.